lib/puppet/type/ios_stp_global.rb

```diff
@@ -48,36 +52,48 @@
       desc:    'Max hops value for the spanning tree',
     },
     mst_name:         {
-      type:      'Optional[String]',
-      desc:      'Configuration name.',
+      type:    'Optional[String]',
+      desc:    'Configuration name.',
     },
     mst_priority:         {
       type:    'Optional[Array[Tuple[String,Integer]]]',
-      desc:      'An array of [mst_inst_list, priority] pairs.',
+      desc:    'An array of [mst_inst_list, priority] pairs.',
     },
     mst_revision:         {
-      type:      'Optional[Integer]',
-      desc:      'Configuration revision number.',
+      type:    'Optional[Integer]',
+      desc:    'Configuration revision number.',
     },
     pathcost:      {
       type:    'Optional[Enum["long","short"]]',
       desc:    'Method to calculate default port path cost',
     },
+    portfast:       {
```

Comment:
Managing `portfast` only seems to support adding and removing a value from the array will not remove it but the run will detect it needs to make a change, i.e.:

```
Notice: /Stage[main]/Main/Ios_stp_global[default]/portfast: portfast changed ['default', 'bpduguard_default', 'bpdufilter_default'] to ['bpduguard_default'] (corrective)
```

Might need to do a `no spanning-tree portfast` for each of the 3 values before doing the update, to ensure the correct behaviour. This should also resolve the ordering issue I found